ui/components/component-library/text-field/text-field.types.ts

```diff
@@ -0,0 +1,138 @@
+import { PolymorphicComponentPropWithRef, StyleUtilityProps } from '../box';
+import { InputProps } from '../input';
+import { InputComponent } from '../input/input.types';
+
+export enum TextFieldSize {
+  Sm = 'sm',
+  Md = 'md',
+  Lg = 'lg',
+}
+
+export enum TextFieldType {
+  Text = 'text',
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  Number = 'number',
+  Password = 'password',
+  Search = 'search',
+}
+
+export interface TextFieldStyleUtilityProps
+  extends Omit<StyleUtilityProps, 'type'> {
+  /**
+   * Autocomplete allows the browser to predict the value based on earlier typed values
+   */
+  autoComplete?: boolean;
+  /**
+   * If `true`, the input will be focused during the first mount.
+   */
+  autoFocus?: boolean;
+  /**
+   * An additional className to apply to the text-field
+   */
+  className?: string;
+  /**
+   * The default input value, useful when not controlling the component.
+   */
+  defaultValue?: string | number;
+  /**
+   * If `true`, the input will be disabled.
+   */
+  disabled?: boolean;
+  /**
+   * If `true`, the input will indicate an error
+   */
+  error?: boolean;
+  /**
+   * The id of the `input` element.
+   */
+  id?: string;
+  /**
+   * The the component that is rendered as the input
+   * Defaults to the Text component
+   */
+  InputComponent?: InputComponent;
+  /**
+   * Attributes applied to the `input` element.
+   */
+  inputProps?: InputProps<'span'>;
+  /**
+   * Component to appear on the left side of the input
+   */
+  startAccessory?: React.ReactNode;
+  /**
+   * Component to appear on the right side of the input
+   */
+  endAccessory?: React.ReactNode;
+  /**
+   * Use inputRef to pass a ref to the html input element.
+   */
+  inputRef?:
+    | React.RefObject<HTMLInputElement>
+    | ((instance: HTMLInputElement | null) => void);
+  /**
+   * Max number of characters to allow
+   */
+  maxLength?: number;
+  /**
+   * Name attribute of the `input` element.
+   */
+  name?: string;
+  /**
+   * Callback fired on blur
+   */
+  onBlur?: () => void;
+  /**
+   * Callback fired when the value is changed.
+   */
+  onChange?: () => void;
+  /**
+   * Callback fired when the TextField is clicked on
+   */
+  onClick?: () => void;
+  /**
+   * Callback fired on focus
+   */
+  onFocus?: () => void;
```

Comment:
it would be better to add typesafety here?

Reply:
Agreed! Is there specific types for these event handlers that we can add?

Reply:
we can do this?
```
onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
onClick?: (event: React.MouseEvent<HTMLInputElement>) => void;
onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
```